From 56fd7016dc3ab6a5960c77aa839dc74279d30b3c Mon Sep 17 00:00:00 2001
From: Drew Moseley <drew.moseley@northern.tech>
Date: Tue, 21 May 2019 17:16:03 +0000
Subject: [PATCH] imx6ul-var-dart: Find splash screen on active rootfs with
 Mender.

Note that this patch is hardcoded to the Mender defines that are
expected. Given when this code executes, the logic in mender_setup has
not run yet so we are just hard coding the values here.

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>
---
 .../variscite/mx6ul_var_dart/mx6ul_var_dart.c | 26 ++++++++++++-------
 include/configs/mx6ul_var_dart.h              |  1 -
 2 files changed, 16 insertions(+), 11 deletions(-)

diff --git a/board/variscite/mx6ul_var_dart/mx6ul_var_dart.c b/board/variscite/mx6ul_var_dart/mx6ul_var_dart.c
index 041a5cd05d..f44dab5c5f 100644
--- a/board/variscite/mx6ul_var_dart/mx6ul_var_dart.c
+++ b/board/variscite/mx6ul_var_dart/mx6ul_var_dart.c
@@ -483,15 +483,15 @@ size_t display_count = ARRAY_SIZE(displays);
 #ifdef CONFIG_SPLASH_SCREEN
 static void set_splashsource_to_boot_rootfs(void)
 {
-	if (!env_check("splashsourceauto", "yes"))
-		return;
-
 #ifdef CONFIG_NAND_BOOT
-	env_set("splashsource", "nand");
+	if (env_check("mender_uboot_if", "mmc"))
+		setenv("splashsource", "emmc");
+        else
+		setenv("splashsource", "nand");
 #else
-	if (mmc_get_env_dev() == 0)
+	if (env_check("mender_uboot_dev", "0"))
 		env_set("splashsource", "sd");
-	else if (mmc_get_env_dev() == 1)
+	else
 		env_set("splashsource", "emmc");
 #endif
 }
@@ -503,12 +503,18 @@ int splash_screen_prepare(void)
 	if (!is_cpu_type(MXC_CPU_MX6ULZ)) {
 		char sd_devpart_str[5];
 		char emmc_devpart_str[5];
+		char ubi_devpart_str[15], ubi_part;
 		u32 sd_part, emmc_part;
 
-		sd_part = emmc_part = env_get_ulong("mmcrootpart", 10, 0);
+		sd_part = emmc_part = env_get_ulong("mender_boot_part", 10, 0);
+		if (sd_part == 0)
+			ubi_part = 'a';
+		else
+			ubi_part = 'b';
 
-		sprintf(sd_devpart_str, "0:%d", sd_part);
-		sprintf(emmc_devpart_str, "1:%d", emmc_part);
+		snprintf(sd_devpart_str, 5, "0:%d", sd_part);
+		snprintf(emmc_devpart_str, 5, "1:%d", emmc_part);
+		snprintf(ubi_devpart_str, 15, "ubi0:rootfs%c", ubi_part);
 
 		struct splash_location var_splash_locations[] = {
 			{
@@ -528,7 +534,7 @@ int splash_screen_prepare(void)
 				.storage = SPLASH_STORAGE_NAND,
 				.flags = SPLASH_STORAGE_FS,
 				.mtdpart = "rootfs",
-				.ubivol = "ubi0:rootfs",
+				.ubivol = ubi_devpart_str,
 			},
 		};
 
diff --git a/include/configs/mx6ul_var_dart.h b/include/configs/mx6ul_var_dart.h
index 75b8dab240..26e3e9943c 100644
--- a/include/configs/mx6ul_var_dart.h
+++ b/include/configs/mx6ul_var_dart.h
@@ -124,7 +124,6 @@
 	"fdt_high=0xffffffff\0" \
 	"initrd_high=0xffffffff\0" \
 	"panel=VAR-WVGA-LCD\0" \
-	"splashsourceauto=yes\0" \
 	"splashfile=/boot/splash.bmp\0" \
 	"splashimage=0x83100000\0" \
 	"splashenable=setenv splashfile /boot/splash.bmp; " \
-- 
2.21.0

